app/src/main/java/com/space/quiz_app/presentation/quiz_home_screen/ui/QuizHomeFragment.kt

```diff
@@ -0,0 +1,87 @@
+package com.space.quiz_app.presentation.quiz_home_screen.ui
+
+import androidx.core.view.isVisible
+import com.space.quiz_app.R
+import com.space.quiz_app.common.extensions.collectFlow
+import com.space.quiz_app.common.extensions.viewBinding
+import com.space.quiz_app.databinding.QuizHomeFragmentBinding
+import com.space.quiz_app.presentation.base.fragment.QuizBaseFragment
+import com.space.quiz_app.presentation.quiz_home_screen.adapter.QuizSubjectsAdapter
+import com.space.quiz_app.presentation.quiz_home_screen.log_out_dialog.LogOutDialog
+import com.space.quiz_app.presentation.quiz_home_screen.view_model.QuizHomeViewModel
+import kotlin.reflect.KClass
+
+class QuizHomeFragment : QuizBaseFragment<QuizHomeViewModel>() {
+
+    private val binding by viewBinding(QuizHomeFragmentBinding::bind)
+
+    override val layout: Int
+        get() = R.layout.quiz_home_fragment
+
+    override val viewModelClass: KClass<QuizHomeViewModel>
+        get() = QuizHomeViewModel::class
+
+    private val subjectsAdapter by lazy {
+        QuizSubjectsAdapter()
+    }
+
+    override fun onCreateFragment() {
+        viewModel.getUsername()
+    }
+
+    override fun onBind() {
+        observe()
+        initRecycler()
+        logOut()
+        setNavigation()
+    }
+
+    private fun observe() {
+        collectFlow(viewModel.usernameState) {
+            binding.greetingTextView.text =
+                String.format(getString(R.string.hello_user), it)
```

Comment:
write extension

Reply:
for what? I don't use such function anywhere else and writing extension takes more code than this